docs/clients/dotnet/21.2/migration-to-gRPC.md

```diff
@@ -625,6 +625,248 @@ You should be careful in defining the retry policy. Not all operations are by de
+    position,
+    filter,
+    filteredSettings,
+    EventAppeared,
+    LiveProcessingStarted,
+    SubscriptionDropped
+);
+```
+
+You need to use:
+
+```csharp
+var filter = 
+    new SubscriptionFilterOptions (EventTypeFilter.ExcludeSystemEvents());
+
+await grpcClient.SubscribeToAllAsync(
+    position,
+    EventAppeared,
+    false,
+    SubscriptionDropped,
+    filter
+);
+```
+
+`CatchUpSubscriptionFilteredSettings` and `Filter` classes from the TCP client were merged into single `SubscriptionFilterOptions` in gRPC.
+
+Read more in the [gRPC server-side filtering docs](/clients/grpc/subscriptions.md#server-side-filtering).
+
+### Knowing when live processing started
+
+TCP client provides the possibility to provide a handler that will be called when live processing starts:
+
+```csharp{6}
+tcpConnection.SubscribeToStreamFrom(
+    streamName,
+    streamPosition,
+    false,
+    EventAppeared,
+    subscription => Console.WriteLine("Processing live"),
+    SubscriptionDropped
+);
+```
+
+gRPC clients do not provide such a feature. However, you can handle that by comparing the current stream (or the `$all` stream) position with the position of the last received event from the subscription. You also need to define threshold as if the system is alive, then new events will constantly appear, and you might not get into the situation where you're fully caught up (especially for the `$all` stream).
+
+You can perform gap measurement in a way as below:
+
+```csharp
+public static class GapMeasurement
+{
+    public static async Task<ulong> GetStreamSubscriptionGap
+    (
+        this EventStoreClient grpcClient,
+        string streamName,
+        StreamPosition subscriptionLastPosition,
+        CancellationToken cancellationToken
+    )
+    {
+        var getCurrentLastEvent = await grpcClient
+            .ReadStreamAsync(
+                Direction.Backwards,
+                streamName,
+                StreamPosition.End,
+                1,
+                resolveLinkTos: false,
+                cancellationToken: cancellationToken
+            ).LastAsync(cancellationToken);
+
+        var currentLastPosition = getCurrentLastEvent.Event.EventNumber.ToUInt64();
+
+        return subscriptionLastPosition - currentLastPosition;
+    }
+
+    public static async Task<ulong> GetAllStreamSubscriptionGap
+    (
+        this EventStoreClient grpcClient,
+        Position subscriptionLastPosition,
+        CancellationToken cancellationToken
+    )
+    {
+        var getCurrentLastEvent = await grpcClient
+            .ReadAllAsync(
+                Direction.Backwards,
+                Position.End,
+                1,
+                resolveLinkTos: false,
+                cancellationToken: cancellationToken
+            ).LastAsync(cancellationToken);
+
+        var currentLastPosition = getCurrentLastEvent.Event.Position.CommitPosition;
+
+        return subscriptionLastPosition.CommitPosition - currentLastPosition;
```

Comment:
Careful as this is not a gap in number of events , but indirectly in number of bytes

Reply:
Good catch 👍 I added a note in https://github.com/EventStore/documentation/pull/498/commits/533a42f79130bf695ab4657b0c00b21570b2c11c.